stdlib/public/runtime/Enum.cpp

```diff
@@ -348,17 +348,25 @@ static void storeMultiPayloadValue(OpaqueValue *value,
 #if defined(__BIG_ENDIAN__)
   unsigned numPayloadValueBytes =
       std::min(layout.payloadSize, sizeof(payloadValue));
-  memcpy(bytes,
+  memcpy(bytes + sizeof(OpaqueValue *) - numPayloadValueBytes,
          reinterpret_cast<char *>(&payloadValue) + 4 - numPayloadValueBytes,
          numPayloadValueBytes);
+  if (layout.payloadSize > sizeof(payloadValue) &&
+      layout.payloadSize > sizeof(OpaqueValue *)) {
+    memset(bytes, 0,
+           sizeof(OpaqueValue *) - numPayloadValueBytes);
+    memset(bytes + sizeof(OpaqueValue *), 0,
+           layout.payloadSize - sizeof(OpaqueValue *));
+  }
 #else
   memcpy(bytes, &payloadValue,
          std::min(layout.payloadSize, sizeof(payloadValue)));
-#endif
+
   // If the payload is larger than the value, zero out the rest.
   if (layout.payloadSize > sizeof(payloadValue))
     memset(bytes + sizeof(payloadValue), 0,
            layout.payloadSize - sizeof(payloadValue));
+#endif
```

Comment:
Thanks for the review.

The code from lines 365-368 are still available when `__BIG_ENDIAN__` is not defined (as they are between `#else` and `#endif`).  Nothing has changed for little endian platforms.

The Interpreter/enum.swift test was failing on s390x and is now passing with this change.  We didn't observe any regression in other tests with this change.

@garyliu1, FYI